### PR TITLE
Add blog to website

### PR DESCRIPTION
**Pull request by welteki**
## Description

Add blog functionality to the actuated website. The app was converted from create react app to a Next.js application for this.

Post can be added as markdown files with frontmatter in the `posts` folder.

An RSS feed for the blog is generated.

## Motivation and Context

Extend actuated site with a blog section.

## How Has This Been Tested?

Verified the main functionality is working by adding some sample post.

- Verified twitter end youtube embeds in posts are working.
- Verified syntax highlighting for code sections.
- Verified the correct metadata for social sharing is added for posts.
- Verified the RSS feed is generated.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I've read the [CONTRIBUTION](https://github.com/openfaas/faas/blob/master/CONTRIBUTING.md) guide
- [x] I have signed-off my commits with `git commit -s`
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,70 +1,21 @@
-# Getting Started with Create React App
-
-This project was bootstrapped with [Create React App](https://github.com/facebook/create-react-app).
+# Usage
 
 ## Available Scripts
 
 In the project directory, you can run:
```
Comment (alexellis) [inline README.md:5]: Let's add the commands to install the "next" CLI?

File: pages/_app.js [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+import '../styles/globals.css'
+
+// See: https://nextjs.org/docs/advanced-features/measuring-performance#web-vitals
+export function reportWebVitals(metric) {
+  if (metric.label === 'web-vital') {
+    // console.log(metric) // The metric object ({ id, name, startTime, value, label }) is logged to the console
+  }
+}
+
+export default function MyApp({ Component, pageProps }) {
```
Comment (alexellis) [inline pages/_app.js:11]: I suppose we could change `MyApp`

File: pages/blog/[id].js [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+            </p>
+            <div className="flex text-sm leading-5 text-gray-500">
+              <Date dateString={post.date} />
+              <span className="mx-1"></span>
+            </div>
+          </div>
+        </div>
+        <div
+          className="blog-max-w prose prose-med text-gray-500 mx-auto"
+          dangerouslySetInnerHTML={{ __html: post.contentHtml }}
```
Comment (alexellis) [inline pages/blog/[id].js:57]: Out of interest, is there a way to use tailwind CSS classes on the generated HTML? Or does all styling for the blog have to be done separately from Tailwind using modifiers on the elements?

File: pages/blog/[id].js [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+          key="og_title"/>
+        <meta
+          property="og:description"
+          content={post.description}
+          key="og_description" />
+      </Head>
+      <div className="py-16 bg-white overflow-hidden">
+      <div className="px-4 sm:px-6 lg:px-8">
+        <div className="text-lg max-w-prose mx-auto mb-6">
+          <h1 className="mt-2 mb-8 text-3xl text-center leading-8 font-extrabold tracking-tight text-gray-900 sm:text-4xl sm:leading-10">
```
Comment (alexellis) [inline pages/blog/[id].js:37]: I'd expect an ID on the most important tags like this for customisation via CSS.

File: pages/blog/[id].js (hunk 3 above)
Comment (welteki) [reply pages/blog/[id].js:57]: I used the [tailwind typography plugin](https://tailwindcss.com/docs/typography-plugin). It offers a couple of ways to modify the styling. One is using [tailwind element modifiers](https://tailwindcss.com/docs/typography-plugin#element-modifiers). It is also possible to provide [custom css via `tailwind.config.js`](https://tailwindcss.com/docs/typography-plugin#customizing-the-css).

I don't know if it is possible to use tailwind classes on the generated html is would have to look into that.

File: pages/blog/[id].js (hunk 4 above)
Comment (alexellis) [reply pages/blog/[id].js:37]: @welteki thoughts?

File: posts/blazing-fast-ci-with-microvms.md [hunk 5]
```diff
@@ -0,0 +1,250 @@
...
+title: Blazing fast CI with MicroVMs
+description: I saw an opportunity to fix self-hosted runners for GitHub Actions. Actuated is now in pilot and aims to solve most if not all of the friction.
+author: Alex Ellis
+tags:
+- cicd
+- bare-metal
+- kubernetes
+- DevOps
+- linux
+- fireckracker
```
Comment (alexellis) [inline posts/blazing-fast-ci-with-microvms.md:11]: Typo

File: posts/blazing-fast-ci-with-microvms.md [hunk 6]
```diff
@@ -0,0 +1,250 @@
+---
+title: Blazing fast CI with MicroVMs
```
Comment (alexellis) [inline posts/blazing-fast-ci-with-microvms.md:2]: There's some kind of "og URL" that we can include in posts like this where it's copied from another website.

I think we'll need that in place before we merge.

Like:

`<link rel="canonical" href="http://﻿﻿﻿silvernailwebdesign.com/reposting-articles"/﻿﻿﻿>﻿﻿﻿﻿`

https://silvernailwebdesign.com/reposting-articles/


```yaml
canonical: https://blog.alexellis.io/blazing-fast-ci-with-microvms/
```

**Comment by alexellis** [comment]
What command do I need to test this locally? Should it be in the README file?

File: posts/blazing-fast-ci-with-microvms.md (hunk 6 above)
Comment (welteki) [reply posts/blazing-fast-ci-with-microvms.md:2]: Setting `canonical` in the post frontmatter is now supported.

**Comment by welteki** [comment]
> What command do I need to test this locally? Should it be in the README file?

The available script and what they do are already in the README. To start a local Next.js development server you can run: `npm run dev`

**Pull request merged**